File: leetcode/graph/83. Remove Duplicates from Sorted List/py/main.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

# Time O(N)
# Space O(1)
class Solution:
    def deleteDuplicates(self, head):
        p = head
        
        while p:
            a = p.next
            
            while a and p.val == a.val:
                a = a.next
                
            p.next = a
            p = p.next
        
        return head
            

File: leetcode/graph/83. Remove Duplicates from Sorted List/js/main.js
// Time O(N)
// Space O(1)
const deleteDuplicates = head => {
  let a = head;

  while (a && a.next) {
    if (a.val == a.next.val) {
      a.next = a.next.next;
    } else {
      a = a.next;
    }
  }

  return head;
};
